haskell/haskell.bzl

```diff
@@ -617,27 +619,58 @@ def _build_haskell_lib(
                 ),
             )
 
-            link.add(objects)
-
             infos = get_link_args_for_strategy(
                 ctx,
                 nlis,
                 to_link_strategy(link_style),
             )
             link.add(cmd_args(unpack_link_args(infos), prepend = "-optl"))
 
-            ctx.actions.run(
-                link,
-                category = "haskell_link" + artifact_suffix.replace("-", "_"),
+
+
+        if(ctx.attrs.use_response_file_for_linker):
```

Comment:
The Haskell toolchain already has a setting for enabling a resonse file, [see here](https://github.com/MercuryTechnologies/buck2-prelude/blob/3c86203c48791f6491211a1a9236da7931f6b257/haskell/compile.bzl#L562-L570). Maybe we could re-use that same toolchain property here and enable the attribute in our toolchain target.

Reply:
oh. great. I didn't know about that `use_argsfile`. I will check that out. thanks 🙏 !

Reply:
I just found that the current implementation of `use_argsfile` is not compatible with our module-level GHC invocation in fact because it uses the same argument file for each module compilation and makes runtime error due to the conflict. So anyway we need to change it with proper module-level filename change (but in general using argsfile for each module compilation does not makes much sense). 
Since we anyway need to have different implementation for `do_link`, I am going to have a separate attribute and impl as I was doing for now. Later, let's unify these `use_argsfile` more properly. I think that flag as toolchain level is also not a good idea. (some big `haskell_library` makes sense with argsfile, but for small one, it's just additional complexity.)

Reply:
but I am renaming `use_response_file_for_linker` to `use_argsfile_at_link` (in haskell_library attrs) to conform the naming convention. :-) 
I am making the change more minimal.